### DISCUSSION TITLE
MIN 4600TL-XH with ARK 10.2 battery System / Eastron SDM 630 Smartmeter Data / Using Grott 3.0

### DISCUSSION THREAD

**Discussion by aberse** [johanmeijer/grott#617] [Q&A]
Good Morning, 
at first many thanks for your work!
Grott 3.0 is working well but i missing some data from the SDM 630 Smartmeter.
At the Web from Growatt i could see, the today's energie consuption and some more informations. 
![image](https://github.com/user-attachments/assets/a91b0534-51d5-43d8-ad60-25b9a57b2586)

But the data telegram i get from the Smartmeter has less informations.
T060120
T06NN20
	 - Growatt new layout processing
		 - record layout :  T06NN20

	 - Grott data record date/time processing started
	 - date-time:  2024-12-30T08:00:11
	 - Grott values retrieved:
		 -  datalogserial        :  GZL0DA6283
		 -  pvserial             :  KMM0CLC07J
		 -  voltage_l1           :  232.9
		 -  voltage_l2           :  232.6
		 -  voltage_l3           :  232.8
		 -  Current_l1           :  0.2
		 -  Current_l2           :  0.4
		 -  Current_l3           :  2.1
		 -  act_power_l1         :  57.3
		 -  act_power_l2         :  16.4
		 -  act_power_l3         :  288.0
		 -  app_power_l1         :  62.2
		 -  app_power_l2         :  111.5
		 -  app_power_l3         :  496.2
		 -  react_power_l1       :  -23.4
		 -  react_power_l2       :  -110.3
		 -  react_power_l3       :  -399.4
		 -  powerfactor_l1       :  926
		 -  powerfactor_l2       :  146
		 -  powerfactor_l3       :  593
		 -  pos_rev_act_power    :  361.7
		 -  pos_act_power        :  361.7
		 -  rev_act_power        :  361.7
		 -  app_power            :  662.5
		 -  react_power          :  -530.9
		 -  powerfactor          :  559
		 -  frequency            :  50.0
		 -  L1-2_voltage         :  403.1
		 -  L2-3_voltage         :  403.0
		 -  L3-1_voltage         :  403.2
		 -  pos_act_energy       :  209.4
		 -  rev_act_energy       :  5.8
	 - MQTT jsonmsg: 
2024-12-30 08:00:16,286 - grottserver - 	DEBUG: process_data_record initiated
2024-12-30 08:00:16,286 - grottserver - 	DEBUG: data record process ended with renturncode: 0

Are those informations not includet in the data or not processed?
I attache the protokoll and an excel export of the data.
[grott.log](https://github.com/user-attachments/files/18272621/grott.log)
[GZL0DA6283_1 Electric-meter data - 2024-12-30_2024-12-30.xls](https://github.com/user-attachments/files/18272622/GZL0DA6283_1.Electric-meter.data.-.2024-12-30_2024-12-30.xls)

**Comment by johanmeijer** [comment]
I will look into that. I know that at the growatt website some values are calculated. I am not sure if the information is in the SDM record or it is calculated. There might be an active energy today and that can be positive or negative (I know that is true for some other values). In that case Growatt adds an additional field that only is being filled if the data is positive. 

There are some possibilities in Grott to create this kind of fields but I think I only have enabled this for Inverter data (I am not sure anymore, sometimes you loose the oversight of your own code ;) ),

**Comment by johanmeijer** [comment]
There is more information in the record: 

Can you specify in the [Generic] section of the grott.ini

    includeall = True 
     
 and look is you see: 
 
      pos_act_energy_kvar  
      rev_act_energy_kvar  
      app_energy_kvar      
      act_energy_kwh       
      react_energy_kvar    

I do not see the positive active (reverse active) today. Can that be calculated ones from pos_act_energy that are maybe overtime totals. Then you have to set a day begin value and substract that from the actual.  
I do not think that the SDM630 keeps daily values (it is not aware of day/time). At this moment I am not able to calculate this in Grott.

**Reply by aberse** [reply]
Hi johan,
thank you for your work and help.
I think you are right and there is no calculated "today load consuption" or anythink else. 
So i help myself and i took your grottcsv.py and change it to a grottmysql.py.
The data of the inverter is saved every minute and i calculate the consuption from every timestamped consuption-data to next.   
If you are interested in the grottmysql.py  i can send you.
For every one who has the problem with the power hysterese up and down around the min SOC, i created a linux bash with is working with grott 3.0. It is working with your grottcsv.py and my grottmysql.py
It analyses the data (soc, powerin, priority, systemstate etc..) and on the "right" time i switch the system into standby. 
Maybe you can include it in grott i was not able to do :(

check_bms.sh checks the state and switch the system into standby 
grottmysql.py is for copy the data to mysql and creates the datafields and table in the database.
  
[check_bms.zip](https://github.com/user-attachments/files/18481688/check_bms.zip)
[grotmysql.zip](https://github.com/user-attachments/files/18481731/grotmysql.zip)
![image](https://github.com/user-attachments/assets/3a9f143d-8322-468a-bbf1-db81ade5b9b8)
